src/test/java/io/split/android/client/storage/attributes/AttributesStorageImplTest.java

```diff
@@ -0,0 +1,200 @@
+package io.split.android.client.storage.attributes;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class AttributesStorageImplTest {
+
+    @Mock
+    private PersistentAttributesStorage persistentAttributesStorage;
+    private AttributesStorageImpl attributesStorage;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        attributesStorage = new AttributesStorageImpl(persistentAttributesStorage);
+    }
+
+    @Test
+    public void loadLocalFetchesValuesFromPersistentStorageIfPersistentStorageIsNotNull() {
+        attributesStorage.loadLocal();
+
+        Mockito.verify(persistentAttributesStorage).getAll();
+    }
+
+    @Test
+    public void loadLocalDoesNotInteractWithPersistentStorageWhenItIsNull() {
+        attributesStorage = new AttributesStorageImpl(null);
+
+        attributesStorage.loadLocal();
+    }
+
+    @Test
+    public void clearCallsClearInPersistentStorageIfPersistentStorageIsNotNull() {
+        attributesStorage.clear();
+
+        Mockito.verify(persistentAttributesStorage).clear();
+    }
+
+    @Test
+    public void clearDoesNotCallClearInPersistentStorageWhenItIsNull() {
+        attributesStorage = new AttributesStorageImpl(null);
+
+        attributesStorage.clear();
+    }
+
+    @Test
+    public void getFetchesCorrespondingValuePreviouslyLoadedFromPersistentStorageIfPersistentStorageIsNotNull() {
+        Map<String, Object> defaultValuesMap = getDefaultValuesMap();
+
+        Mockito.when(persistentAttributesStorage.getAll()).thenReturn(defaultValuesMap);
+
+        attributesStorage.loadLocal();
+
+        assertEquals("value1", attributesStorage.get("key1"));
```

Comment:
Perhaps checking also value count?

Reply:
Yes, good catch. I should be checking for all values actually.